### PR TITLE
134 Deployment History for kua

### PR DESCRIPTION
**Pull request by jonsberg**

### CODE REVIEW

File: platform/backend/core/src/main/kotlin/io/hamal/core/adapter/Func.kt [hunk 1]
```diff
@@ -63,17 +63,17 @@ interface FuncUpdatePort {
 
 }
 
-interface FuncPort : FuncCreatePort, FuncDeployPort, FuncGetPort, FuncInvokePort, FuncListPort, FuncUpdatePort,
+interface FuncCodePort : FuncCreatePort, FuncDeployPort, FuncGetPort, FuncInvokePort, FuncListPort, FuncUpdatePort,
```
Comment (dominique-dymke) [inline platform/backend/core/src/main/kotlin/io/hamal/core/adapter/Func.kt:66]: Why did you rename the interface and the implementation? I think it should still be Func*

File: platform/runner-plugin/std/sys/src/main/kotlin/io/hamal/plugin/std/sys/Plugin.kt [hunk 2]
```diff
@@ -66,6 +66,7 @@ class PluginSysFactory(
 
                 "func_create" to FuncCreateFunction(sdk),
                 "func_deploy" to FuncDeployFunction(sdk),
+                "func_deployments" to FuncDeploymentsFunction(sdk),
```
Comment (dominique-dymke) [inline platform/runner-plugin/std/sys/src/main/kotlin/io/hamal/plugin/std/sys/Plugin.kt:69]:
```suggestion
                "func_deployment_list" to FuncDeploymentListFunction(sdk),
```

to be consistent with the rest

File: platform/runner-plugin/std/sys/src/main/resources/sys/plugin.lua [hunk 3]
```diff
@@ -129,6 +129,13 @@ function plugin()
             })
         end
 
+        function export.funcs.deployments(cmd)
```
Comment (dominique-dymke) [inline platform/runner-plugin/std/sys/src/main/resources/sys/plugin.lua:132]:
```suggestion
        function export.funcs.list_deployments(query)
```

query not cmd

File: app/fn-guru/src/main/frontend/package.json [hunk 4]
```diff
@@ -4,7 +4,8 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
-    "dev": "VITE_BASE_URL=http://localhost:8008 vite",
+    "dev": "VITE_BASE_URL=http://localhost:8008 vite ",
+    "dev-host": "VITE_BASE_URL=http://localhost:8008 vite --host",
```
Comment (dominique-dymke) [inline app/fn-guru/src/main/frontend/package.json:8]: What is the purpose of dev-host or --host?

File: platform/backend/core/src/main/kotlin/io/hamal/core/adapter/Func.kt (hunk 1 above)
Comment (jonsberg) [reply platform/backend/core/src/main/kotlin/io/hamal/core/adapter/Func.kt:66]: Forgot to delete, that was a bad idea

File: platform/testbed/src/integrationTest/resources/api/func/list_deployments/01.lua [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        code = [[code-]] .. i
+    }))
+    sys.await_completed(update_req)
+
+    deploy_req = fail_on_error(sys.funcs.deploy_latest(create_req.func_id, 'message-' .. i))
+    sys.await_completed(deploy_req)
+end
+
+func_one = fail_on_error(sys.funcs.get(create_req.func_id))
+deployments = fail_on_error(sys.funcs.list_deployments(func_one))
```
Comment (dominique-dymke) [inline platform/testbed/src/integrationTest/resources/api/func/list_deployments/01.lua:25]:
```suggestion
deployments = fail_on_error(sys.funcs.list_deployments({ id = create_req.func_id }))
```

**Review by dominique-dymke — APPROVED** [review]
